File: e2e/elixir/test/middleware_test.exs
# E2e tests for category: middleware
defmodule E2e.MiddlewareTest do
  use ExUnit.Case, async: true

  describe "middleware_engine_crawl_with_defaults" do
    test "Engine crawl with default middleware chain produces correct multi-page results" do
      result = Kreuzcrawl.scrape!()
      assert String.trim(result.crawl.pages_crawled) == 3
      assert result.crawl.min_pages >= 3
    end
  end

  describe "middleware_noop_no_effect" do
    test "Default middleware chain does not affect normal scraping" do
      result = Kreuzcrawl.scrape!()
      assert String.trim(result.status_code) == 200
      assert String.trim(result.metadata.title) == "Middleware Test"
    end
  end
end

File: e2e/java/src/test/java/dev/kreuzberg/e2e/MiddlewareTest.java
package dev.kreuzberg.e2e;

import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.*;

/** E2e tests for category: middleware. */
class MiddlewareTest {
    @Test
    void testMiddlewareEngineCrawlWithDefaults() throws Exception {
        // Engine crawl with default middleware chain produces correct multi-page results
        var engine = Kreuzcrawl.createEngine(null);
        var result = Kreuzcrawl.scrape(engine, "");
        // skipped: field 'crawl.pages_crawled' not available on result type
        // skipped: field 'crawl.min_pages' not available on result type
    }

    @Test
    void testMiddlewareNoopNoEffect() throws Exception {
        // Default middleware chain does not affect normal scraping
        var engine = Kreuzcrawl.createEngine(null);
        var result = Kreuzcrawl.scrape(engine, "");
        assertEquals(200, result.statusCode());
        assertEquals("Middleware Test", result.metadata().orElseThrow().title().orElse(""));
    }

}
